spec/models/order_item_spec.rb

```diff
@@ -113,4 +117,48 @@
 
     it_behaves_like "#mark_item"
   end
+
+  describe '#can_order?' do
+    context 'when process_scope is applicable' do
+      before { order_item.update(:process_scope => 'applicable') }
+
+      it 'is orderable' do
+        order_item.update(:state => 'Approved')
+        expect(order_item.can_order?).to be_truthy
+      end
+
+      it 'is not orderable' do
+        order_item.update(:state => 'Created')
+        expect(order_item.can_order?).to be_falsey
+      end
+    end
+
+    context 'when process_scope is before' do
+      before { order_item.update(:process_scope => 'before') }
+
+      it 'is orderable' do
+        order_item.update(:state => 'Created')
+        expect(order_item.can_order?).to be_truthy
+      end
+
+      it 'is not orderable' do
+        order_item.update(:state => 'Ordered')
+        expect(order_item.can_order?).to be_falsey
+      end
+    end
+  end
+
+  context 'when process_scope is after' do
+    before { order_item.update(:process_scope => 'after') }
+
+    it 'is orderable' do
+      order_item.update(:state => 'Created')
+      expect(order_item.can_order?).to be_truthy
+    end
+
+    it 'is not orderable' do
+      order_item.update(:state => 'Completed')
+      expect(order_item.can_order?).to be_falsey
+    end
+  end
```

Comment:
I think this can be cleaned up a bit to take advantage of rspec a bit better. Something like this:
```ruby
  before do
    order_item.update(:process_scope => process_scope, :state => state)
  end

  context 'when process_scope is applicable' do
    let(:process_scope) { "applicable" }

    context "when the state is approved" do
      let(:state) { "Approved" }

      it 'is orderable' do
        expect(order_item.can_order?).to be_truthy
      end
    end

    context "when the state is created" do
      let(:state) { "Created" }

      it 'is not orderable' do
        expect(order_item.can_order?).to be_falsey
      end
    end
  end

  # And then likewise for the 'before' and 'after' process scopes
```

It also reads a bit better if you take the whole sentence into account: "when process_scope is applicable (and) when state is created (it) is orderable" versus how you have it now you will have:
"when process_scope is applicable (it) is orderable" and
"when process_scope is applicable (it) is not orderable", which sounds directly contradictory.